### PR TITLE
fix(bigtable): set timeouts for asynchronous operations

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #4926.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6014)

### CODE REVIEW

**Comment by coryan** [comment]
@PeteyPii FYI...

File: google/cloud/internal/setup_context.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Copyright 2020 Google LLC
```
Comment (devbww) [inline google/cloud/internal/setup_context.h:1]: s/2020/2021/

File: google/cloud/internal/async_retry_loop_test.cc [hunk 2]
```diff
@@ -262,6 +262,56 @@ TEST(AsyncRetryLoopTest, ExhaustedDuringBackoff) {
...
+      .WillOnce(Return(false));
+  EXPECT_CALL(*mock, IsExhausted)
+      .WillOnce(Return(false))
+      .WillOnce(Return(false))
+      .WillOnce(Return(false))
+      .WillRepeatedly(Return(true));
+  EXPECT_CALL(*mock, IsPermanentFailure).WillRepeatedly(Return(false));
+  EXPECT_CALL(*mock, Setup).Times(3);
+
+  std::vector<ms> sleep_for;
```
Comment (devbww) [inline google/cloud/internal/async_retry_loop_test.cc:298]: Unused (and so `ms` unused too)?

**Review by devbww — APPROVED** [review]
Thanks.

File: google/cloud/internal/setup_context.h (hunk 1 above)
Comment (coryan) [reply google/cloud/internal/setup_context.h]: Done.

File: google/cloud/internal/async_retry_loop_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/internal/async_retry_loop_test.cc]: Fixed.

**Pull request merged**